crates/rpc/src/handlers/block.rs

```diff
@@ -200,6 +210,24 @@ pub async fn get_beacon_block_from_id(
         })
 }
 
+pub async fn get_filtered_block_tree(db: &ReamDB) -> Result<Vec<BeaconBlock>, ApiError> {
+    let justified_checkpoint = db
+        .justified_checkpoint_provider()
+        .get()
+        .map_err(|_| ApiError::InternalError)?;
+    let root = justified_checkpoint.root;
+
+    let mut blocks = HashMap::with_hasher(RandomState::default());
+    let store = Store { db: db.clone() };
+
+    store.filter_block_tree(root, &mut blocks).map_err(|err| {
```

Comment:
`filter_block_tree` gives us a tree of blocks near the head of the chain. Currently you are just returning all the block roots to the RPC, don't we need to walk the tree and only return all the leaves right?

Reply:
I read through pseudocode in `consensus-specs` and code in `filter_block_tree` and the implementation is pretty much the same. I think we can use from this one.

Reply:
You can use `filter_block_tree`, but you need to filter out all the blocks that are parents of leaves, which isn't done yet. Currently the whole subtree returned by `filter_block_tree` is returned to the RPC.

``Retrieves all possible chain heads (leaves of fork choice tree).``

`filter_block_tree()` returns the fork choice tree, but this PR is currently not filtering the tree returned by `filter_block_tree()` for the `leaves/heads`